architect/systems/spectrometers.py

```diff
@@ -125,17 +124,18 @@ def get_signal(wavelength):
 
         return signal
 
-    def get_noise(self):
+    def get_noise(self, wavelength, radiance: LUT):
         """Get the noise.
-
         Ref: https://www.notion.so/utat-ss/Noise-21ff532ac4334fbeab4aabf6372c9848
-
         """
         noise = np.sqrt(
-            (signal.decompose() * unit.electron)
+            (
+                self.get_signal(wavelength=wavelength, radiance=radiance).decompose()
+                * unit.electron
+            )
             + self.sensor.get_n_bin() * (self.sensor.get_dark_noise() * unit.pix) ** 2
             + self.sensor.get_quantization_noise() ** 2
-            + self.sensor.get_n_bin() * self.sensor.get_noise_read() ** 2
+            + self.sensor.get_n_bin() * self.sensor.get_noise(self.get_signal(wavelength=wavelength, radiance=radiance)) ** 2
```

Comment:
The function to be called here for this parameter should be `sensor.get_dark_shot_noise()`. I recently split and moved some functions around for dark noise.
```suggestion
            + self.sensor.get_n_bin() * self.sensor.get_dark_shot_noise()
```

Reply:
There is also another error with this function where there isn't a 'get_dark_noise' function anymore in line 138. Should I be replacing the 'get_dark_noise' function with 'get_dark_shot_noise'? Or should I do as your suggestion states and replace the 'get_noise' function with 'get_dark_shot_noise'?

Reply:
Just an update. I changed the equation around a little bit to look like the equation in the mathematical models for net noise. I did not use sensor.get_dark_shot_noise() because the equation looks like its asking for sigma(read) (manufacturer given value) not sigma(dark) (output of sensor.get_dark_shot_noise()). Let me know if I'm understanding this correctly.